scripts/sync-bottle-hashes.sh

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env bash
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# This script takes the new tag and the OS name as its arguments
+# and handles git shenanigans related to syncing two pipelines over a common branch.
+
+if [ -z "$1" ] && [ -z "$2" ]; then
+    git config user.name "Serokell" # necessary for pushing
+    git config user.email "hi@serokell.io" # TODO: right values
+    git fetch
+
+    branch_name="auto/update-brew-formulae-$1"
+
+    function compute_and_commit {
+        ./scripts/bottle-hashes.sh .
+        git commit -a -m "[Chore] Add $1 hashes to brew formulae"
+    }
+
+   pr_body="Add bottle hashes for $1
+
+Problem: we have built brew bottles for the new Octez release, but their hashes
+aren't in the formulae yet.
+
+Solution: added the hashes.
+"
+
+    # Git doesn't have an easy way to check out a branch regardless of whether it exists.
```

Comment:
I think that there's an easier way to do this syncing using `checkout`.

The flow should be something like this:
1. `git fetch --all`
2. `git checkout <branch_name>`
    Note that by default if a remote `<branch_name>` exists this will both create the local branch and start tracking the remote one. If the remote one doesn't exist it will fail.
3. Iff (2) failed then do `git checkout -b <branch_name>` and `git push -u <remote> <branch_name>`.
    Note that in the remote case in which both push at the same time `git push` won't fail (because they are pushing the same thing).
4. At this point we know that both remote and local branch exist and are synced (for now at least).
5.  `compute_and_commit` here
6. `git push`, or at least try to.
     If this fails it means that someone else pushed on the branch in the meanwhile.
7. Iff (6) fails, run `git fetch --all` and `git reset --hard <remote>/<branch_name>` then jump back to (4)

Reply:
Unless I'm missing something, this version loses the notion of whether the PR is already opened. It's not necessary a big deal, because GitHub doesn't allow opening two PRs from one branch, but it is, in my opinion, not as clean: we now have to open the PR from the first pipeline that finishes, meaning it will have only half of the work for an hour or so. It also means opening the PR will fail half of the tries.

But it is shorter, so I did like you suggested :/

Reply:
Oh I was pretty sure I left a comment about opening the PR, looks like I forgot or it got lost :disappointed: 

Anyway, yes, inevitably we'll rely on one to fail when the PR already exists, and do nothing there.